src/storage/src/table/state_table.rs

```diff
@@ -113,171 +116,109 @@ impl<S: StateStore> StateTable<S> {
         Ok(())
     }
 
-    pub async fn iter(&self, epoch: u64) -> StorageResult<StateTableRowIter<'_, S>> {
-        let mem_table_iter = self.mem_table.buffer.iter().peekable();
-        StateTableRowIter::new(
+    pub async fn iter(
+        &self,
+        epoch: u64,
+    ) -> StorageResult<Pin<Box<impl Stream<Item = StorageResult<Option<Row>>> + Send + '_>>> {
+        let mem_table_iter = self.mem_table.buffer.iter();
+        Ok(Box::pin(StateTableRowIter::into_stream(
```

Comment:
I think we should let users to pin this. Just return `StateTableRowIter::into_stream` should be enough.